node/node.go

```diff
@@ -653,6 +653,25 @@ func (node *AlgorandFullNode) GetPendingTransaction(txID transactions.Txid) (res
 		// Keep looking in the ledger.
 	}
 
+	// quick check for confirmed transactions with LastValid in future
+	// this supposed to cover most of the cases where REST checks for the most recent txns
+	if r, confirmed := node.ledger.CheckConfirmedTail(txID); confirmed {
```

Comment:
Connecting to main goal of PR, this is our short circuit from looking into blocks when we have a transaction in the pool. We now only look at blocks if transaction is not found in the pool.